azurerm/internal/services/network/express_route_circuit_connection_resource.go

```diff
@@ -0,0 +1,272 @@
+	if err != nil {
+		return err
+	}
+
+	expressRouteCircuitConnectionParameters := network.ExpressRouteCircuitConnection{
+		Name: utils.String(id.ConnectionName),
+		ExpressRouteCircuitConnectionPropertiesFormat: &network.ExpressRouteCircuitConnectionPropertiesFormat{
+			AddressPrefix: utils.String(d.Get("address_prefix_ipv4").(string)),
+			ExpressRouteCircuitPeering: &network.SubResource{
+				ID: utils.String(circuitPeeringId.ID()),
+			},
+			PeerExpressRouteCircuitPeering: &network.SubResource{
+				ID: utils.String(circuitPeerPeeringId.ID()),
+			},
+		},
+	}
+
+	if v, ok := d.GetOk("authorization_key"); ok {
+		expressRouteCircuitConnectionParameters.ExpressRouteCircuitConnectionPropertiesFormat.AuthorizationKey = utils.String(v.(string))
+	}
+
+	if v, ok := d.GetOk("address_prefix_ipv6"); ok {
+		expressRouteCircuitConnectionParameters.ExpressRouteCircuitConnectionPropertiesFormat.Ipv6CircuitConnectionConfig = &network.Ipv6CircuitConnectionConfig{
+			AddressPrefix: utils.String(v.(string)),
+		}
+	}
+
+	future, err := client.CreateOrUpdate(ctx, id.ResourceGroup, id.ExpressRouteCircuitName, id.PeeringName, id.ConnectionName, expressRouteCircuitConnectionParameters)
+	if err != nil {
+		return fmt.Errorf("creating %s: %+v", id, err)
+	}
+
+	if err := future.WaitForCompletionRef(ctx, client.Client); err != nil {
+		return fmt.Errorf("waiting for creation of %s: %+v", id, err)
+	}
+
+	d.SetId(id.ID())
+
+	return resourceExpressRouteCircuitConnectionRead(d, meta)
+}
+
+func resourceExpressRouteCircuitConnectionRead(d *pluginsdk.ResourceData, meta interface{}) error {
+	client := meta.(*clients.Client).Network.ExpressRouteCircuitConnectionClient
+	ctx, cancel := timeouts.ForRead(meta.(*clients.Client).StopContext, d)
+	defer cancel()
+
+	id, err := parse.ExpressRouteCircuitConnectionID(d.Id())
+	if err != nil {
+		return err
+	}
+
+	resp, err := client.Get(ctx, id.ResourceGroup, id.ExpressRouteCircuitName, id.PeeringName, id.ConnectionName)
+	if err != nil {
+		if utils.ResponseWasNotFound(resp.Response) {
+			log.Printf("[INFO] %s does not exist - removing from state", *id)
+			d.SetId("")
+			return nil
+		}
+		return fmt.Errorf("retrieving %s: %+v", *id, err)
+	}
+
+	d.Set("name", id.ConnectionName)
+	d.Set("peering_id", parse.NewExpressRouteCircuitPeeringID(id.SubscriptionId, id.ResourceGroup, id.ExpressRouteCircuitName, id.PeeringName).ID())
+
+	if props := resp.ExpressRouteCircuitConnectionPropertiesFormat; props != nil {
+		d.Set("address_prefix_ipv4", props.AddressPrefix)
+
+		authorizationKey := ""
+		if props.AuthorizationKey != nil {
+			authorizationKey = *props.AuthorizationKey
+		}
+		d.Set("authorization_key", authorizationKey)
+
+		addressPrefixIPv6 := ""
+		if props.Ipv6CircuitConnectionConfig != nil && props.Ipv6CircuitConnectionConfig.AddressPrefix != nil {
+			addressPrefixIPv6 = *props.Ipv6CircuitConnectionConfig.AddressPrefix
+		}
+		d.Set("address_prefix_ipv6", addressPrefixIPv6)
```

Comment:
Can be simplified to:

```suggestion
		if props.Ipv6CircuitConnectionConfig != nil  {
			d.Set("address_prefix_ipv6", props.Ipv6CircuitConnectionConfig.AddressPrefix )
		}
```

Reply:
@magodo , we cannot do this because we cannot set it as empty string when Ipv6CircuitConnectionConfig  is nil.

Reply:
That's true. Thank you!

Reply:
Then can we also ensure to set the peering id in the following code?

Reply:
Sure. Updated